src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -44,16 +47,18 @@ public class ScriptHostManager : IScriptHostEnvironment, IDisposable
         private ILogger _logger;
 
         private ScriptSettingsManager _settingsManager;
-        private AutoRecoveringFileSystemWatcher _scriptFileWatcher;
         private CancellationTokenSource _restartDelayTokenSource;
 
         public ScriptHostManager(ScriptHostConfiguration config, IScriptEventManager eventManager = null, IScriptHostEnvironment environment = null)
             : this(config, ScriptSettingsManager.Instance, new ScriptHostFactory(), eventManager, environment)
         {
             if (config.FileWatchingEnabled)
             {
-                _scriptFileWatcher = new AutoRecoveringFileSystemWatcher(config.RootScriptPath);
-                _scriptFileWatcher.Changed += OnScriptFileChanged;
+                // We only setup a subscription here as the actual ScriptHost will create the publisher
+                // when initialized.
+                _fileEventSubscription = EventManager.OfType<FileEvent>()
```

Comment:
Subscribing to file events will be very common in our codebase - you're adding a few in this PR. We might consider an extension method(s) to facilitate common patterns, allowing these to be simplified to EventManager.OnFileEvent(OnScriptFileChanged).

Just mentioning this as a future suggestion.